src/main/java/com/lgtm/easymoney/services/impl/SearchServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package com.lgtm.easymoney.services.impl;
+
+import com.lgtm.easymoney.exceptions.ResourceNotFoundException;
+import com.lgtm.easymoney.models.User;
+import com.lgtm.easymoney.payload.SearchRsp;
+import com.lgtm.easymoney.repositories.UserRepository;
+import com.lgtm.easymoney.services.SearchService;
+import com.lgtm.easymoney.repositories.AccountRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import java.util.regex.Pattern;
+
+@Service
+public class SearchServiceImpl implements SearchService {
+    private final UserRepository userRepository;
+    private final AccountRepository accountRepository;
+
+    @Autowired
+    public SearchServiceImpl(UserRepository userRepository, AccountRepository accountRepository) {
+        this.userRepository = userRepository;
+        this.accountRepository = accountRepository;
+    }
+    
+    @Override
+    public User getUserByID(Long id) {
```

Comment:
integrate these `getUserByXX` into userService and call api via the userService.
question is : some of the fields are not unique, maybe email? how to handle this properly? potentially a list of users....

the search service should be robust imho, thus i think we need to refactor the response and handle 1) one client 2) no result 3) multiple users match

Reply:
Emails are unique. The same email cannot be registered twice. 

For the integration of these methods, I will gauge other team members' interests. Please refer to the slack channel.